models/ed/tests/testthat/test.met2model.R

```diff
@@ -11,7 +6,7 @@ test_that("Met conversion runs without error", {
   in.prefix <- "CRUNCEP"
   start_date <- "2000-01-01"
   end_date <- "2000-12-31"
-  result <- met2model.ED2(in.path, in.prefix, outfolder, start_date, end_date)
+  result <- met2model.ED2(in.path, in.prefix, tempdir(), start_date, end_date)
```

Comment:
Is it enough to start with `outdir <- tempdir()` and end with `unlink(outdir, recursive = TRUE)` outside of the tests?

Reply:
Is it enough to start with `outdir <- tempdir()` and end with `unlink(outdir, recursive = TRUE)` outside of the tests?

Reply:
Looks like the better way is with `withr::defer()`?

``` r
test_that("Met conversion runs without error", {
  outdir <- tempdir()
  withr::defer(unlink(outdir, recursive = TRUE))
  
  ...

  result <- met2model.ED2(in.path, in.prefix, outdir, start_date, end_date)
  expect_s3_class(result, "data.frame")
})
```
Either way it looks like this test is still kinda broken because `met2model.ED2()` prints an error as a character string that's not detected and then calls a bunch of `testthat` functions (#2987)

Reply:
Yep, `withr::defer` is a good approach, but be careful to use `tempfile()` (returns a path that's unique to this call but doesn't exist yet; do manually create and delete it) rather than `tempdir()` (returns a path that already exists but is shared by whole session; don't manually delete it).